### PR TITLE
fix alignment of chunk gates

### LINKED ISSUE #5181: r-oldrel-windows-ix86+x86_64 vignette error
This WARN has just started today on CRAN with 1.14.2. But 1.14.2 is an utterly minimal patch of 1.14.0 so I doubt it's related to 1.14.2 per se. My guess is that r-oldrel checks run when the package is published on CRAN, but don't run again thereafter. So I'm thinking that something else is different on r-oldrel between the last publish date and now. The version of r-oldrel may well have changed in that time, for example. We have been seeing several fails on CRAN's windows recently due to packages not being available (like knitr) unrelated to data.table; those tend to get fixed by CRAN within a week or so.  But this one (`attempt to use zero-length variable name`) seems like it could be a data.table problem, perhaps under R 4.0.5.

```
using R version 4.0.5 (2021-03-31)
using platform: x86_64-w64-mingw32 (64-bit)
using session charset: ISO8859-1
checking for file 'data.table/DESCRIPTION' ... OK
this is package 'data.table' version '1.14.2'
checking package namespace information ... OK
checking package dependencies ... OK
checking if this is a source package ... OK
checking if there is a namespace ... OK
checking for hidden files and directories ... OK
checking for portable file names ... OK
checking whether package 'data.table' can be installed ... OK
checking installed package size ... OK
checking package directory ... OK
checking 'build' directory ... OK
checking DESCRIPTION meta-information ... OK
checking top-level files ... OK
checking for left-over files ... OK
checking index information ... OK
checking package subdirectories ... OK
checking R files for non-ASCII characters ... OK
checking R files for syntax errors ... OK
loading checks for arch 'i386'
checking whether the package can be loaded ... OK
checking whether the package can be loaded with stated dependencies ... OK
checking whether the package can be unloaded cleanly ... OK
checking whether the namespace can be loaded with stated dependencies ... OK
checking whether the namespace can be unloaded cleanly ... OK
checking loading without being on the library search path ... OK
checking use of S3 registration ... OK
loading checks for arch 'x64'
checking whether the package can be loaded ... OK
checking whether the package can be loaded with stated dependencies ... OK
checking whether the package can be unloaded cleanly ... OK
checking whether the namespace can be loaded with stated dependencies ... OK
checking whether the namespace can be unloaded cleanly ... OK
checking loading without being on the library search path ... OK
checking use of S3 registration ... OK
checking dependencies in R code ... OK
checking S3 generic/method consistency ... OK
checking replacement functions ... OK
checking foreign function calls ... OK
checking R code for possible problems ... [25s] OK
checking Rd files ... OK
checking Rd metadata ... OK
checking Rd cross-references ... OK
checking for missing documentation entries ... OK
checking for code/documentation mismatches ... OK
checking Rd \usage sections ... OK
checking Rd contents ... OK
checking for unstated dependencies in examples ... OK
checking line endings in shell scripts ... OK
checking line endings in C/C++/Fortran sources/headers ... OK
checking line endings in Makefiles ... OK
checking compilation flags in Makevars ... OK
checking for GNU extensions in Makefiles ... OK
checking for portable use of $(BLAS_LIBS) and $(LAPACK_LIBS) ... OK
checking use of PKG_*FLAGS in Makefiles ... OK
checking pragmas in C/C++ headers and code ... OK
checking compiled code ... OK
checking installed files from 'inst/doc' ... OK
checking files in 'vignettes' ... OK
checking examples ...
running examples for arch 'i386' ... [7s] OK
running examples for arch 'x64' ... [8s] OK
checking for unstated dependencies in 'tests' ... OK
checking tests ...
running tests for arch 'i386' ... [199s] OK
  Running 'autoprint.R' [0s]
  Comparing 'autoprint.Rout' to 'autoprint.Rout.save' ... OK
  Running 'froll.R' [24s]
  Running 'knitr.R' [1s]
  Comparing 'knitr.Rout' to 'knitr.Rout.save' ... OK
  Running 'main.R' [172s]
  Running 'nafill.R' [2s]
  Running 'other.R' [0s]
  Running 'types.R' [0s]
running tests for arch 'x64' ... [222s] OK
  Running 'autoprint.R' [0s]
  Comparing 'autoprint.Rout' to 'autoprint.Rout.save' ... OK
  Running 'froll.R' [23s]
  Running 'knitr.R' [1s]
  Comparing 'knitr.Rout' to 'knitr.Rout.save' ... OK
  Running 'main.R' [195s]
  Running 'nafill.R' [2s]
  Running 'other.R' [0s]
  Running 'types.R' [0s]
checking for unstated dependencies in vignettes ... OK
checking package vignettes in 'inst/doc' ... OK
checking re-building of vignette outputs ... [41s] WARNING
Error(s) in re-building vignettes:
--- re-building 'datatable-benchmarking.Rmd' using rmarkdown
--- finished re-building 'datatable-benchmarking.Rmd'

--- re-building 'datatable-faq.Rmd' using rmarkdown
--- finished re-building 'datatable-faq.Rmd'

--- re-building 'datatable-importing.Rmd' using rmarkdown
--- finished re-building 'datatable-importing.Rmd'

--- re-building 'datatable-intro.Rmd' using rmarkdown
Loading required package: data.table
--- finished re-building 'datatable-intro.Rmd'

--- re-building 'datatable-keys-fast-subset.Rmd' using rmarkdown
Loading required package: data.table
--- finished re-building 'datatable-keys-fast-subset.Rmd'

--- re-building 'datatable-reference-semantics.Rmd' using rmarkdown
Loading required package: data.table
Quitting from lines 135-154 (datatable-reference-semantics.Rmd)
Error: processing vignette 'datatable-reference-semantics.Rmd' failed with diagnostics:
attempt to use zero-length variable name
--- failed re-building 'datatable-reference-semantics.Rmd'

--- re-building 'datatable-reshape.Rmd' using rmarkdown
Loading required package: data.table
--- finished re-building 'datatable-reshape.Rmd'

--- re-building 'datatable-sd-usage.Rmd' using rmarkdown
Loading required package: data.table
--- finished re-building 'datatable-sd-usage.Rmd'

--- re-building 'datatable-secondary-indices-and-auto-indexing.Rmd' using rmarkdown
Loading required package: data.table
--- finished re-building 'datatable-secondary-indices-and-auto-indexing.Rmd'

SUMMARY: processing the following file failed:
  'datatable-reference-semantics.Rmd'

Error: Vignette re-building failed.
Execution halted
checking PDF version of manual ... OK
DONE
Status: 1 WARNING
```

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #5181

cc @mattdowle in case you were already trying to debug

### CODE REVIEW

**Comment by yihui** [comment]
Oh I didn't realize you already submitted a PR (I was doing the same thing in https://github.com/yihui/data.table/commit/a23b512d8782e7d7109d619330f6be7facdd8de2).

**Comment by mattdowle** [comment]
@yihui Why didn't you give me any notice? Did you run revdep checks? CRAN runs revdep checks -- what did you respond when they asked if you had liaised with broken revdep maintainers?
I have only just published data.table v1.14.2 in the last week, and it took 5 elapsed days to get done. It is extremely annoying to have such breakage without you contacting me.

**Comment by mattdowle** [comment]
@yihui can you request the update be rolled back so you can give proper notice? I don't want to have data.table in WARN status on every CRAN instance. CRAN then puts me under pressure to clear the errors.  I need time to manage releases. You should have given me notice as per CRAN policy.

**Comment by yihui** [comment]
@mattdowle I totally understand your feeling. I just explained in email and also at https://github.com/Rdatatable/data.table/issues/5181#issuecomment-929829761. I'm very sorry about it, and will work on a patch release soon. Thanks for your patience!

**Comment by yihui** [comment]
The new version of knitr was submitted last night and appeared on CRAN early this morning. I'll keep an eye on the [CRAN check results of data.table](https://cran.r-project.org/web/checks/check_results_data.table.html). Hopefully the vignette problem will disappear later today, and you won't need to worry about it anymore.

File: vignettes/datatable-reference-semantics.Rmd [hunk 1]
```diff
@@ -92,7 +92,7 @@ It can be used in `j` in two ways:
     # when you have only one column to assign to you
     # can drop the quotes and list(), for convenience
     DT[, colA := valA]
-	  ```
+    ```
```
Comment (yihui) [inline vignettes/datatable-reference-semantics.Rmd:95]: @MichaelChirico I discovered a really weird thing here and I wonder if you could help me understand it. The fix you made here was correct, but if you download the source package from CRAN: https://cran.r-project.org/src/contrib/data.table_1.14.8.tar.gz you will find that the first two spaces are actually a tab character `\t`.

**Pull request merged**